hotkdump/core/hotkdump.py

```diff
@@ -493,17 +503,22 @@ def run(self):
                     print(f"{key}={self.kdump_file.vmcoreinfo.get(key)}")
                 return
 
-            extracted_vmlinux = None
+            dbg_type = self.debug_file_type()
+            vmlinux_ddeb = self.params.debug_file if dbg_type == "ddeb" else None
+            extracted_vmlinux = self.params.debug_file if dbg_type == "vmlinux" else None
 
-            if not self.params.no_debuginfod:
-                extracted_vmlinux = self.maybe_download_vmlinux_via_debuginfod()
+            # Avoid downloading debug symbols if a debug file was specified.
+            if not dbg_type:
+                if not self.params.no_debuginfod:
+                    extracted_vmlinux = self.maybe_download_vmlinux_via_debuginfod()
 
-            if not extracted_vmlinux and not self.params.no_pullpkg:
-                vmlinux_ddeb = self.maybe_download_vmlinux_via_pullpkg()
-                if vmlinux_ddeb == "":
-                    logging.error("vmlinux ddeb dowload failed.")
-                    return
+                if not extracted_vmlinux and not self.params.no_pullpkg:
+                    vmlinux_ddeb = self.maybe_download_vmlinux_via_pullpkg()
+                    if vmlinux_ddeb == "":
+                        logging.error("vmlinux ddeb dowload failed.")
+                        return
```

Comment:
This part can be much simpler if we do the following:

- make a function named `maybe_get_user_specified_vmlinux` and return the vmlinux (automatically extract if .ddeb)
- change `maybe_download_vmlinux_via_pullpkg` to return a path to extracted vmlinux (move extract_vmlinux_ddeb call to maybe_download_vmlinux_via_pullpkg)
- move self.params.no_debuginfod and self.params.no_pullpkg checks to respective maybe_download_vmlinux*  functions and return None when True

... then we can simply do the following:

```python
extracted_vmlinux = self.maybe_get_user_specified_vmlinux() or self.maybe_download_vmlinux_via_debuginfod() or self.maybe_download_vmlinux_via_pullpkg()
```